crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -304,6 +318,60 @@ where
             _block: PhantomData,
         })
     }
+
+    /// Returns the last archived segment if available.
+    /// If there is no archived segment, reconstructs the genesis segment if it contains `requested_piece_index`.
+    ///
+    /// If there is a last archived segment, `requested_piece_index` is ignored.
+    pub fn last_archived_segment(
```

Comment:
Refactoring like extracting a method should ideally be in a separate commit so I don't need to review it, trying to understand what you have changed (if anything).

Reply:
Thanks, I’ll split that commit out today.